### PR TITLE
fix(components/dialog): backport ConfirmDialog enhancement

### PR DESCRIPTION
**Pull request by acateland**
**What is the problem this PR is trying to solve?**
see https://github.com/Talend/ui/pull/484
**What is the chosen solution to this problem?**
see https://github.com/Talend/ui/pull/484

All specific css for this feature put into ConfirmDialog css module was ported to the bootstrap theme and both ConfirmDialog and Dialog use it

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/theme.sasslint.txt [hunk 1]
```diff
@@ -23,13 +23,22 @@ src/theme/_forms.scss
   316:9   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
   457:8   warning  Vendor prefixes should not be used                       no-vendor-prefixes
 
+src/theme/_modals.scss
+  10:3  warning  Expected indentation of 1 tab but found 2   indentation
```
Comment (talend-sromain) [inline output/theme.sasslint.txt]: Some lint here

File: packages/theme/src/theme/_modals.scss [hunk 2]
```diff
@@ -5,12 +5,42 @@
 	flex-direction: column;
 }
 
+/** IE-FIX */
+.modal-dialog {
```
Comment (jsomsanith) [inline packages/theme/src/theme/_modals.scss]: Can you merge with the previous class ? Those 2 blocks are for `.modal-dialog` and the second one overrides some properties of the first one.

File: packages/theme/src/theme/_modals.scss [hunk 3]
```diff
@@ -5,12 +5,42 @@
...
+	display: flex;
+	flex-direction: row;
+	flex-wrap: nowrap;
+
+	>.modal-content {
```
Comment (jsomsanith) [inline packages/theme/src/theme/_modals.scss]: This is supposed to style the same elements as the `.modal-content` below. Can you merge them ?

File: packages/theme/src/theme/_modals.scss [hunk 4]
```diff
@@ -25,10 +55,17 @@
 	}
 
 	&-body {
+		flex-basis: auto;
 		flex-grow: 1;
+		flex-shrink: 1;
+		display: flex;
+		flex-direction: column;
+		overflow: hidden;
```
Comment (jsomsanith) [inline packages/theme/src/theme/_modals.scss]: I don't understand why we have those 3 properties above. Why do we style the content that can be whatever we want ?

**Pull request closed without merging**